### DISCUSSION TITLE
Status Update: 2022-10-12

### DISCUSSION THREAD

**Discussion by phil-opp** [dora-rs/dora#102] [Status Update]
In this discussion, we describe our recent progress and the main topics that we would like to discuss. We use a separate post for each topic to keep the discussion focused. Feel free to create new posts for topics that are not mentioned yet.

**Comment by phil-opp** [comment]
## Start creating command-line interface

We implemented the following commands:

- **`dora new`:** Create a new operator or custom node from a template.

  Example:
  
  ```
  dora new --kind operator --lang rust example_operator ./example
  ```
- **`dora graph`:** Visualize a given dataflow as a graph.
- **`dora build`:** Run the specified build steps for all operators/nodes of the given dataflow.
- **`dora check`:** Verify that all nodes/operators exist and that all input-output mappings are valid.
- **`dora start`:** Start running the given dataflow.
- **`dora destroy`:** Stop the dora backround daemons.

See https://github.com/dora-rs/dora/issues/93 for more details.

**Comment by haixuanTao** [comment]
## Tracing for `dora-runtime`

- Merged tracing for dora within `dora-runtime` to trace operators for both `python` and shared libraries. 

## Metrics for `dora-runtime`

- Merged metrics for dora within `dora-runtime`.

**Comment by haixuanTao** [comment]
## Dora-drives 

- Fixing an issue with Lidar to Camera projection for obstacle detection.
- Finished the tutorial for `dora-drives` with one liners for testing.

**Comment by haixuanTao** [comment]
## Looking for recruiting an internship.

- Looking for a 6 month intern for next year in full remote.

**Comment by haixuanTao** [comment]
## Release workflow

Github release workflow: https://github.com/dora-rs/dora/pull/90 with prebuilt package for Windows, Mac and Linux, for all dora binaries as well as the python package installable via `pip install dora-rs`

**Comment by haixuanTao** [comment]
## Issues with Zenoh 0.6.0

Currently facing a silent bug in zenoh: https://github.com/dora-rs/dora/pull/94 that is keeping us from upgrading it
